File: frontend/src/app/_model/product.ts
export class Product{
  constructor(id: number,
              productCategory: string,
              productName: string,
              productAmount: number,
              productPrice: number,
              productDiscount: number,
              productDate: Date,
              productDescription: string,
              productStatus: boolean) {
    this.id = id;
    this.productCategory = productCategory;
    this.productName = productName;
    this.productAmount = productAmount;
    this.productPrice = productPrice;
    this.productDiscount = productDiscount;
    this.productDate = productDate;
    this.productDescription = productDescription;
    this.productStatus = productStatus;
  }

  id: number;
  productCategory: string;
  productName: string;
  productAmount: number;
  productPrice: number;
  productDiscount: number;
  productDate: Date;
  productDescription: string;
  productStatus: boolean;
}

File: backend/src/main/java/com/eshop/backend/dao/models/Product.java
package com.eshop.backend.dao.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Product {
    private Long id;

    private int productCategory;

    private String productName;

    private int productAmount;

    private double productPrice;

    private double productDiscount;

    private Date productDate;

    private String productPict;

    private String productDescription;

    private String productStatus;

}
